### PR TITLE
✨ Bump CAPI to v1.8.1, Helm to v3.15.3, and Go version to 1.22.5

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,7 @@ SHELL:=/usr/bin/env bash
 #
 # Go.
 #
-GO_VERSION ?= 1.21.5
+GO_VERSION ?= 1.22.5
```
Comment (dtzar) [inline Makefile:26]: FWIW - 1.22.6 is latest 1.22.x

File: go.mod [hunk 2]
```diff
@@ -11,17 +13,17 @@ require (
...
 	go.uber.org/mock v0.4.0
 	gopkg.in/yaml.v2 v2.4.0
-	helm.sh/helm/v3 v3.14.4
-	k8s.io/api v0.29.3
-	k8s.io/apimachinery v0.29.3
-	k8s.io/cli-runtime v0.29.3
-	k8s.io/client-go v0.29.3
-	k8s.io/component-base v0.29.3
-	k8s.io/klog/v2 v2.110.1
+	helm.sh/helm/v3 v3.15.3
```
Comment (dtzar) [inline go.mod:16]: FWIW 3.15.4 is latest

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Comment by mboersma** [comment]
/hold
For squash

**Pull request merged**